### PR TITLE
Fix Linux runner and installation

### PR DESCRIPTION
**Pull request by niezbop**
* Runner now makes sure that the directory containing the log exists
before attempting to create it.

* The UnityProject now parses correctly its version under Linux

* LinuxInstallation uses the correct path for the exe

### CODE REVIEW

File: lib/u3d/unity_runner.rb [hunk 1]
```diff
@@ -36,6 +36,7 @@ def run(installation, args, raw_logs: false)
         log_file = installation.default_log_file
       end
 
+      Utils.ensure_dir File.dirname(log_file)
```
Comment (lacostej) [inline lib/u3d/unity_runner.rb]: Is this related to Linux install and running? Or a more general cross platform feature?

**Comment by lacostej** [comment]
I would love some unit testing for the `editor_version` change.

**Comment by niezbop** [comment]
Will provide unit tests!

File: lib/u3d/unity_runner.rb (hunk 1 above)
Comment (niezbop) [inline lib/u3d/unity_runner.rb]: I believe it is more of a cross platform feature, when Unity has never been run on the system. But I faced it with my shinny new Linux so that's why it is there. Should I create a separated PR?

Comment (lacostej) [inline lib/u3d/unity_runner.rb]: I moved it to a separate one. Already merged.

File: lib/u3d/installer.rb [hunk 2]
```diff
@@ -107,7 +107,7 @@ def default_log_file
     end
 
     def exe_path
-      "#{path}/Unity"
+      "#{path}/Editor/Unity"
```
Comment (lacostej) [inline lib/u3d/installer.rb]: I also moved this to a separate PR, already merged.
